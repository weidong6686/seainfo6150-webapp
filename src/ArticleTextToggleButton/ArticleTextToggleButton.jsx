import React from "react";
import styles from "./ArticleTextToggleButton.module.css"

const ArticleTextToggleButton = (props) => {

    return (
        <div>
            <button className={styles.button} onClick={props.onClick}>{props.buttonText}</button>
        </div>
    );
};

export default ArticleTextToggleButton;
