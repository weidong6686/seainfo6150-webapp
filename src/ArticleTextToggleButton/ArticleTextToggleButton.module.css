.button{
    color: white;
    height: 30px;
    width: 90px;
    background: #c92f1b;
    border-radius: 10px;
}