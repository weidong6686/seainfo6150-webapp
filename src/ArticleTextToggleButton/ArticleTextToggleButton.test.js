import React from "react";
import ArticleTextToggleButton from "./ArticleTextToggleButton";
import { fireEvent } from "@testing-library/react";

describe("ArticleTextToggleButton tests", () => {

  it("renders correctly", () => {
    const { container, getByText } = render(<ArticleTextToggleButton buttonText={"Test Button"} />);
    const button = getByText("Test Button");
    fireEvent.click(button);
    expect(container).toMatchSnapshot();
  });
  
});