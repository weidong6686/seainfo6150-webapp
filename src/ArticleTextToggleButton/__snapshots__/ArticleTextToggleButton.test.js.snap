// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleTextToggleButton tests renders correctly 1`] = `
<div>
  <button
    class="button"
  >
    Test Button
  </button>
</div>
`;
